api/server/response_options.proto

```diff
@@ -5,13 +5,31 @@ package nighthawk.server;
 import "google/protobuf/wrappers.proto";
 import "validate/validate.proto";
 import "envoy/api/v2/core/base.proto";
+import "google/protobuf/duration.proto";
 
-// Options that control the test server response.
+message ConcurrencyBasedDelay {
+  // Minimal delay to add to replies.
+  google.protobuf.Duration minimal_delay = 1 [(validate.rules).duration.gte.nanos = 0];
+  // Factor to use when adding latency as concurrency increases.
+  google.protobuf.Duration concurrency_delay_factor = 2 [(validate.rules).duration.gte.nanos = 0];
```

Comment:
This seems to be a linear model for latency. I wonder if ultimately you want something more like a network queuing model where you see a hyperbolic delay as requests back up significantly. OTOH, you also have some in-built delay from the fact that the test server itself will have some queuing effects.

What's the ultimate goal here? To model overload when the test server isn't fully in overload?

Reply:
The goal here is to model response times associated to an overloaded server, without the server actually being overloaded.

What you are saying here sounds like something me and @jmarantz discussed on Slack; we figured the current linear model would suffice for the concrete use-case backing this, but I think what you're proposing would be a more accurate simulation, and it reflects something that was discussed on Slack. 

Maybe we should incorporate the word "linear" in the name here, that way if someone needs a hyperbolic delay there's no ambiguity, and it's clear what happens here. And it would be very easy to add it.

Reply:
Renamed proto to mention 'linear'  to disambiguate with potential future additions/enhancements, and made it clear that we're running a simplified model in the docs in c40e9ce65c82b62a6f9eab47ac31327a7605a6f4

Reply:
One other question while we're here. If the goal is to model an overloaded server without being overloaded, why make use of actual concurrent requests at all? Couldn't you just inject a fake number of concurrent requests and use that to compute delay?

Reply:
if one is testing behaviour of a load balancer running in front of multiple of these servers that are simulating overload, 
having the delay based on actual concurrency may help verify the load balancer's balancing strategy.